front/src/applications/stdcm/utils/index.ts

```diff
@@ -13,13 +16,23 @@ export const getTimesInfoFromDate = (date?: Date) =>
       }
     : undefined;
 
-export const extractMarkersInfo = (pathSteps: StdcmPathStep[]) =>
-  pathSteps.reduce((acc, step) => {
-    if (step.location) {
-      acc.push({
-        coordinates: step.location.coordinates,
-        name: step.location.name,
-      });
-    }
-    return acc;
-  }, [] as MarkerInformation[]);
+export const extractMarkersInfo = (pathSteps: StdcmPathStep[]): MarkerInformation[] =>
+  pathSteps
+    .filter((step) => step.location)
+    .map((step, index, array) => {
+      let pointType;
+
+      if (index === 0) {
+        pointType = MARKER_TYPE.ORIGIN;
+      } else if (index === array.length - 1) {
```

Comment:
Why use array instead of pathSteps here ?
```suggestion
      } else if (index === pathSteps.length - 1) {
```

Reply:
Done